### PR TITLE
Scripts for BPM equalization!

### PR DESCRIPTION
**Pull request by ericonr**

### CODE REVIEW

**Comment by ericonr** [comment]
While some of these scripts are useful for testing, we should eventually join them into a single script that sets *all* properties on a given crate/slot (or all of them), and just always use that. If it's written in python, it's fast enough. Then the remaining scripts can simply be about specific actions (like syncing FOFB, syncing BPMs, or equalizing antennas).

**Comment by ericonr** [comment]
@guilhermerc 

For using these scripts during deployment:

1. `./reset_bpm_antenna_gains`: run after all BPMs have been updated and IOCs have been rebooted
2. `./set_bpm_rates 48k-eq`: it prints `monit_timing_div=` last, use that as the divisor value for the timing clock
3. use Gabriel's acquisition script (reproduced below) to coordinate a synchronized acquisition across the whole ring (needs beam) (FOFB off is ok)
4. `./save_bpm_antennas fofb48k-test.pickle`: save BPM data
5. `./equalize_bpm_antennas fofb48k-test.pickle eabs fofb48k-test-pvs.txt`: generate PV list
6. `./write_pvs fofb48k-test-pvs.txt`: write gains to BPMs (SOFB off)
7. inspect antenna waveforms to see if we picked the right semi-cycle; otherwise change `['Inv', 'Dir']` to `['Dir', 'Inv']` in `equalize_bpm_antennas` and go back to step 5
8. `./set_bpm_rates 48k`

Acquisition script:

```
#!/usr/bin/env python-sirius

import time
import sys
import numpy as np

from mathphys.functions import load_pickle 
from apsuite.commisslib.measure_orbit_stability import OrbitAcquisition

out_file = sys.argv[1]

def acquire_data(orbacq):
    # acquire data
    print('--- acquire orbit data ---')
    orbacq.acquire_data()
    
    # save machine state to data
    print('--- add devices states to data ---')
    state = read_state(orbacq)
    orbacq.data['state'] = state
    
def read_state(orbacq):
    state = dict()
    state['orbacq'] = orbacq.get_initial_state()
    return state


print('Creating connections with BPMs...')
orbacq = OrbitAcquisition(isonline=True)

print('Waiting EPICS connections to finish...')
# check if devices are connected
orbacq.wait_for_connection()
print('orbacq    : ', orbacq.connected)

time.sleep(2)

print('Reading initial timing state...')
init_state = read_state(orbacq)


print('Preparing Timing for acquisitions...')
# prepare timing
orbacq.prepare_timing()

time.sleep(2)


print('Configuring BPMs for acquisition...')
orbacq.params.orbit_acq_rate        = sys.argv[2]
orbacq.params.orbit_timeout         = 150
orbacq.params.orbit_nrpoints_after  = int(sys.argv[3])
orbacq.params.orbit_acq_repeat      = 0
orbacq.params.trigbpm_delay         = 0
orbacq.params.trigbpm_nrpulses      = 1
orbacq.params.event_mode            = 'External'
print(orbacq.params)

time.sleep(2)


print('Sending timing event and acquiring data...')
print('This action may take some time to finish...')
# acquire data
acquire_data(orbacq)

time.sleep(2)


print(f'Saving data to file {out_file:s}.pickle')
# save data
orbacq.save_data(out_file, overwrite=True)

time.sleep(2)

# recover initial state
orbacq.recover_initial_state(init_state['orbacq'])


print('Check if data is consistent')
data = load_pickle(out_file + '.pickle')
print(data['data']['orbx'].shape)
```

File: scripts/python/set_bpm_rates [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+put_pv(bo_harmonic, bo_harmonic_number, wait=False)
+put_pv(bo_fofb, bo_fofb_rate, wait=False)
+put_pv(bo_facq, bo_facq_rate, wait=False)
+put_pv(bo_monit, bo_monit_rate, wait=False)
+put_pv(bo_switching, bo_switching_rate, wait=False)
+
+wait_pv()
+
+monit_timing_div = (monit_rate // tbt_rate) * harmonic_number / 4
```
Comment (guilhermerc) [inline scripts/python/set_bpm_rates]: @ericonr , can't this be already setted on `AS-RaMO:TI-EVG:Clk3MuxDiv-SP`?

Comment (ericonr) [reply scripts/python/set_bpm_rates]: That seems like a good idea. I'll add it!

File: scripts/python/get_bpm_pvs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+pv_names = sys.argv[1:]
+pvs = []
+
+for crate in range(1, 22):
+    for slot in all_bpm_slots[crate_number(crate)]:
+        prefix = get_pv_prefix(crate, slot)
+        pvs.extend((create_pv(prefix + name) for name in pv_names))
+
+wait_for_pv_connection()
+print_pv(pvs)
```
Comment (ericonr) [inline scripts/python/get_bpm_pvs:24]: make it print like `write_pvs` reads it.

**Pull request merged**